data/templates/setup_iso.py

```diff
@@ -0,0 +1,150 @@
+import argparse
+import subprocess
+import os
+import errno
+import os.path
+import tempfile
+import atexit
+import sys
+import shutil
+import re
+import urllib
+
+tmpdir = ''
+verbose=1
+
+# Cleanup
+@atexit.register
+def cleanup_tmp():
+    if tmpdir != '':
+        subprocess.check_call(['umount', tmpdir])
+        os.rmdir(tmpdir)
+
+def get_iso_info(fname):
+    # Access ISO to determine OS type from supported list
+    label1=subprocess.Popen(['isoinfo', '-i', fname, '-d'], stdout=subprocess.PIPE)
+    return label1.stdout.read()
+
+def do_install_os(osname, osver, src, dest):
+    print 'Installing {0} {1} to {2}/{0}/{1}'.format(osname, osver, dest)
+    dstpath=dest + '/' + osname + '/' + osver
+    if os.path.isdir(dstpath):
+        print 'Found existing directory, bailing...'
+        sys.exit(1)
+
+    if osname is 'ESXi':
+        shutil.copytree(src, dstpath)
+
+    if osname is 'RHEL' or osname is 'Centos':
+        shutil.copytree(src, dstpath)
+
+    if osname is 'LiveCD':
+        initrd='initrd.img'
+        vmlinuz='vmlinuz'
+        if os.path.isfile(src+'/isolinux/initrd0.img'):
+            initrd='initrd0.img'
+            vmlinuz='vmlinuz0'
+        mount1 = subprocess.Popen(['mount'], stdout=subprocess.PIPE)
+        mount2 = subprocess.Popen(['grep', src], stdin=mount1.stdout, stdout=subprocess.PIPE)
+        mount3 = subprocess.Popen(['awk', '{print $1}'], stdin=mount2.stdout, stdout=subprocess.PIPE)
+        isoname = mount3.communicate()[0]
+
+        iso_basename = os.path.basename(isoname).strip()
+        iso_dirname = os.path.dirname(isoname).strip()
+
+        os.makedirs(dstpath)
+        
+        syscall = '(cd "'+iso_dirname+'" && echo "'+iso_basename+'" | cpio -H newc --quiet -L -o )'
+        syscall += ' | gzip -9'
```

Comment:
I wonder if it's possible to set PIPEFAIL or something here, that way if any of these steps error, they don't get swallowed by gzip and produce a corrupted initrd image? I've had some painful debugging experiences in the past related to pipelines like this.